### ISSUE TITLE
Unable to export the data to Word or HTML

### ISSUE THREAD

**Issue by Arbelac** [dotraphael/RFL.Microsoft.AD#1]
Hi,

 I am not able to export the output like below.

Powershell Version :

Name             : ConsoleHost
Version          : 7.3.4
InstanceId       : 909cb448-8342-4843-93ca-ae6ee0a83860
UI               : System.Management.Automation.Internal.Host.InternalHostUserInterface
CurrentCulture   : en-US
CurrentUICulture : en-US
PrivateData      : Microsoft.PowerShell.ConsoleHost+ConsoleColorProxy
DebuggerEnabled  : True
IsRunspacePushed : False
Runspace         : System.Management.Automation.Runspaces.LocalRunspace

.\CollectADData.ps1 -SaveTo 'C:\RFL.Microsoft.AD-main\RFL.Microsoft.AD-main'   ->> it works very well.

.\ExportADData.ps1 -SourceData 'C:\RFL.Microsoft.AD-main\RFL.Microsoft.AD-main' -OutputFormat @('Word', 'HTML') -CompanyName 'Contoso' -CompanyWeb 'http://www.contoso.com' -CompanyEmail 'team@contoso.com'   --->> its not working

My output:

05-05-2023 20:57 Current Folder 'C:\RFL.Microsoft.AD-main\RFL.Microsoft.AD-main'
05-05-2023 20:57 All checks completed successful. Starting collecting data for report
05-05-2023 20:57 Report file name: avivasa.local
Document: Cannot process argument transformation on parameter 'ScriptBlock'. Cannot convert the "
        #region style
        DocumentOption -EnableSectionNumbering -PageSize A4 -DefaultFont 'Arial' -MarginLeftAndRight 71 -MarginTopAndBottom 71 -Orientation Portrait
....
..
..

..

05-05-2023 20:57 Exporting report format Word to C:\RFL.Microsoft.AD-main\RFL.Microsoft.AD-main
05-05-2023 20:57 Exporting report format HTML to C:\RFL.Microsoft.AD-main\RFL.Microsoft.AD-main
05-05-2023 20:57 All Data have been collected and saved to ''
05-05-2023 20:57 You can now use the 'ExportADData.ps1' PowerShell script to create the report file
05-05-2023 20:57 *** Ending ***

**Comment by dotraphael** [comment]
Hi Arbelac,

Try using PowerShell 5.1 instead of 7.x core.

The collectaddata works on both as far as my testing goes. Only the export needs 5.1.

I'm working to make sure it works on core 7.x

Let me know how it goes

**Comment by Arbelac** [comment]
Hi,

Now it works. sorry for my fault :) 

My workflow :

I ran below script In PowerShell 5.1.  it works very well. 
1-  .\CollectADData.ps1 -SaveTo 'C:\RFL.Microsoft.AD-main\RFL.Microsoft.AD-main'

2 - Firstly ,  I ran below script In PowerShell 5.1. But , when attempting to run ExportADData.ps1 then I got the following the error message.
convertfrom-json : a parameter cannot be found that matches parameter name 'ashashtable'.

After investigation , I have found an issue related to the PowerShell version. According to the MS , PowerShell 5.1 does not supported AsHashtable parameter in ConvertFrom-Json cmd-let. 

As summary , I am able to export my AD Report as HTML & Word.

Under PowerShell 5.1 :
.\CollectADData.ps1 -SaveTo 'C:\RFL.Microsoft.AD-main\RFL.Microsoft.AD-main' 

Under PowerShell 7.3.4 :

1 -  import-module PScribo -SkipEditionCheck


2 - .\ExportADData.ps1 -SourceData 'C:\RFL.Microsoft.AD-main\RFL.Microsoft.AD-main' -OutputFormat @('Word', 'HTML') -CompanyName 'Contoso' -CompanyWeb 'http://www.contoso.com/' -CompanyEmail 'team@contoso.com'


thanks,
